samtranslator/internal/model/appsync.py

```diff
@@ -25,19 +25,27 @@ class DynamoDBConfigType(TypedDict, total=False):
     DeltaSyncConfig: DeltaSyncConfigType
 
 
+class LogConfigType(TypedDict, total=False):
+    CloudWatchLogsRoleArn: Intrinsicable[str]
+    ExcludeVerboseContent: bool
```

Comment:
This should be `Optional[bool]`?

Reply:
You can import `NotRequired` from https://pypi.org/project/typing-extensions/

Reply:
The only problem I have with `NotRequired` is that I can't instantiate the TypedDict as empty and then add values to it as I create it. My original idea was to create a `Dict[str, Any]` dictionary and then unpack it in another dictionary with the appropriate TypedDict to get static type checking, but that is not a supported behaviour (see https://github.com/python/mypy/issues/4122).

So if we want to use `NotRequired` now, I think we'd need to use a condition to check if the optional key exists, and then we can create a TypedDict with that optional key or a TypedDict without it. That's a lot of extra code IMO that will suck to add for all of the resources I'm creating, especially those with multiple optional values.

Is there any other way we could go about constructing the dictionary to have static type checking that doesn't introduce so much bloat?